src/components/walletconnect/WcInput/index.tsx

```diff
@@ -51,8 +51,51 @@ const WcInput = (): ReactElement => {
       return
     }
 
-    onInput(clipboard)
-  }, [onInput])
+    return clipboard
+  }, [])
+
+  const onPaste = useCallback(async () => {
+    const clipboard = await getClipboard()
+
+    if (clipboard) {
+      onInput(clipboard)
+    }
+  }, [getClipboard, onInput])
+
+  useEffect(() => {
+    if (!navigator || !window) {
+      return
+    }
+
+    const autofillUri = async () => {
+      let isGranted = false
+
+      try {
+        // @ts-expect-error navigator permissions types don't include clipboard
+        const permission = await navigator.permissions.query({ name: 'clipboard-read' })
+        isGranted = permission.state === 'granted'
+      } catch (e) {
+        setError(asError(e))
+        return
+      }
+
+      if (!isGranted) {
+        return
+      }
+
+      const clipboard = await getClipboard()
+
+      if (clipboard?.startsWith('wc:')) {
+        onInput(clipboard)
+      }
+    }
+
+    window.addEventListener('focus', autofillUri)
```

Comment:
Isn't WcInput rendered only when the WC button in the header is clicked?

Reply:
The `SessionManager` [is always mounted](https://github.com/safe-global/safe-wallet-web/blob/focus-wc-uri/src/components/walletconnect/HeaderWidget/index.tsx#L63) (`keepMounted`) in order to [listen for proposals](https://github.com/safe-global/safe-wallet-web/blob/focus-wc-uri/src/components/walletconnect/SessionManager/index.tsx#L62).

We could hoist [the state](https://github.com/safe-global/safe-wallet-web/blob/focus-wc-uri/src/components/walletconnect/SessionManager/index.tsx#L18) but I think it makes the most sense to keep it in `SessionManager`. What do you think?

Reply:
By drilling, in accordance with https://github.com/safe-global/safe-wallet-web/pull/2630#discussion_r1360392904, we no longer need to keep this mounted.